src/Api/ApiResourceRegistrar.php

```diff
@@ -75,24 +89,52 @@ public function resource(string $resourceName, string $controller, array $option
                 }
 
                 foreach ($actions as $action => $i) {
-                    $routes[$action] = ['verb' => $verb, 'route' => '/'.$action];
+                    $routes[$action] = ['verb' => $verb, 'route' => '/' . $action];
                 }
             }
         }
 
+        $mapping = [];
+
+        if ($modelClass) {
+            $modelName = lcfirst($modelName ?? $this->resolveModelClass($modelClass));
+            $mapping[$modelName] = $modelClass;
+        }
+
         foreach ($routes as $action => $opt) {
             $verb = $opt['verb'];
-            $this->api->$verb($resourceName.$opt['route'], [
-                'as' => trim($resourceName.'.'.$action),
-                'uses' => $controller.'@'.$action
+            $route = $opt['route'];
+            if ($modelName) {
+                $route = str_replace('{id}', "{{$modelName}}", $opt['route']);
+            }
+
+            $this->api->$verb($resourceName . $route, [
+                'as' => trim($resourceName . '.' . $action),
+                'uses' => $controller . '@' . $action,
+                'mapping' => $mapping,
             ]);
         }
     }
 
+    /**
+     * Resolves model class name. Ex: App\Models\User -> User.
```

Comment:
IMHO **resolve** has other purposes, here we just *getShortClassName()* and no need to point that we can use it only for model